packages/app-store/make/api/add.ts

```diff
@@ -0,0 +1,39 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import prisma from "@calcom/prisma";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!req.session?.user?.id) {
+    return res.status(401).json({ message: "You must be logged in to do this" });
+  }
+  const appType = "make_automation";
+  try {
+    const alreadyInstalled = await prisma.credential.findFirst({
+      where: {
+        type: appType,
+        userId: req.session.user.id,
+      },
+    });
+    if (alreadyInstalled) {
+      throw new Error("Already installed");
+    }
+    const installation = await prisma.credential.create({
+      data: {
+        type: appType,
+        key: {},
+        userId: req.session.user.id,
+        appId: "make",
+      },
+    });
+    if (!installation) {
```

Comment:
I don't think we are doing any OAuth in there. We are just creating credentials as a means to mark the app installed. For that we have AppDeclarativeHandler. See [this](https://github.com/aar2dee2/cal.com/blob/6bfabe41e5073ff3a9e38d577ef6d72be2d6eca8/packages/app-store/skiff/api/add.ts#L21) as an example

Reply:
That's already checked [here] when `createDefaultInstallation` is used (https://github.com/aar2dee2/cal.com/blob/6bfabe41e5073ff3a9e38d577ef6d72be2d6eca8/apps/web/pages/api/integrations/%5B...args%5D.ts#L12)

Reply:
The job of `createDefaultInstallation` is to just create the installation. Additional checks are done `defaultIntegrationAddHandler`

Reply:
thanks!

edited this.